src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchShareLogDAO.scala

```diff
@@ -0,0 +1,133 @@
+package org.broadinstitute.dsde.firecloud.dataaccess
+
+import java.time.Instant
+
+import org.broadinstitute.dsde.firecloud.FireCloudException
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol.ShareFormat
+import org.broadinstitute.dsde.firecloud.model.ShareLog.Share
+import org.broadinstitute.dsde.workbench.util.health.SubsystemStatus
+import org.elasticsearch.action.admin.indices.create.{CreateIndexRequest, CreateIndexRequestBuilder, CreateIndexResponse}
+import org.elasticsearch.action.admin.indices.exists.indices.{IndicesExistsRequest, IndicesExistsRequestBuilder, IndicesExistsResponse}
+import org.elasticsearch.action.get.{GetRequest, GetRequestBuilder, GetResponse}
+import org.elasticsearch.action.index.{IndexRequest, IndexRequestBuilder, IndexResponse}
+import org.elasticsearch.action.search.{SearchRequest, SearchRequestBuilder, SearchResponse}
+import org.elasticsearch.action.support.WriteRequest.RefreshPolicy
+import org.elasticsearch.client.transport.TransportClient
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.index.query.QueryBuilder
+import org.elasticsearch.index.query.QueryBuilders._
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.util.hashing.MurmurHash3
+import scala.util.{Failure, Success, Try}
+
+trait ShareQueries {
+  def userShares(userId: String): QueryBuilder = termQuery("userId", userId)
+  def userSharesOfType(userId: String, shareType: String) = boolQuery()
+    .must(termQuery("userId", userId))
+    .must(termQuery("shareType", shareType))
+}
+
+/**
+  * DAO that uses ElasticSearch to log and get records of shares.
+  *
+  * @param client      The ElasticSearch client
+  * @param indexName   The name of the target share log index in ElasticSearch
+  * @param refreshMode Using IMMEDIATE - the default - is a performance hit but ensures transactionality
+  *                    of updates across multiple users.
+  *                    TODO would NONE be a better option here?
```

Comment:
yeah, I'd go for NONE here. I don't see any reason this needs to be atomic/immediately available.